specs/agents/tracing-spans.md

```diff
@@ -70,3 +70,27 @@ To handle edge cases where many spans are captured within a single transaction,
 ```
 
 Here's how the limit can be configured for [Node.js](https://www.elastic.co/guide/en/apm/agent/nodejs/current/agent-api.html#transaction-max-spans) and [Python](https://www.elastic.co/guide/en/apm/agent/python/current/configuration.html#config-transaction-max-spans).
+
+### Exit spans
+
+Exit spans are spans that describe a call to an external service,
+such as an outgoing HTTP request or a call to a database.
+
+Exit spans MUST not have child spans that have a different `type` and `subtype`.
+For example, when capturing a span representing a query to Elasticsearch,
+there should not be an HTTP span for the same operation.
+Doing that would make [breakdown metrics](https://github.com/elastic/apm/blob/master/specs/agents/metrics.md#transaction-and-span-breakdown)
+less meaningful,
+as most of the time would be attributed to `http` instead of `elasticsearch`.
+
+Agents MAY add information from the lower level transport to the exit span, though.
+For example, the HTTP `context.http.status_code` may be added to an `elasticsearch` span.
+
+Exit spans MAY have child spans that have the same `type` and `subtype`.
+For example, an HTTP exit span may have child spans with the `action` `request`, `response`, `connect`, `dns`.
+These spans MUST NOT have any destination context, so that there's no effect on destination metrics.
+
+Agents don't need to prevent the creation of exit child spans via the public API.
+Agents MAY implement internal APIs to create an exit span (such as `Span::createExitSpan`) that flags a span as an exit span (`exit=true`).
```

Comment:
The Python agent (and maybe Ruby/Node.js?) has the concept of a leaf span. If a span is created with `leaf=True`, any child spans are simply dropped. This has been around since forever, and has some overlap with this proposed `exit` flag, and we mostly (maybe exclusively, I'm not 100% sure) use it to drop HTTP spans for instrumented calls that use HTTP as the transport layer.

`leaf` is part of our public API, so changing it to `exit` would involve quite some backwards-compatibility work. Not sure if that's worth the trouble.

Reply:
That's definitely not the intention. It's just an example of how an internal API MAY be implemented.

Reply:
I've updated the section to hopefully make it a bit clearer